### ISSUE TITLE
Conflicting rules: `no-reversed-links` and no-`bare-urls`

### ISSUE THREAD

**Issue by arvigeus** [DavidAnson/markdownlint#268]
This:

```md
(Incorrect link syntax)[https://www.example.com/]
```

Gets corrected to this:

```md
(Incorrect link syntax)[<https://www.example.com/]>
```

I think `bare-urls` should not attempt to fix it in this case.

**Comment by DavidAnson** [comment]
The rules detect and report independently. When attempting to fix both at once, the library will not apply overlapping edits and so the first fix it sees “wins”.

That said, I may be able to detect this scenario and maybe not report it as a bare URL to avoid this scenario? I’m not sure I love that option; I will think about this some more.
